pkg/lint/linter/config.go

```diff
@@ -50,6 +50,11 @@ type Config struct {
 	Deprecation *Deprecation
 }
 
+func (lc *Config) Default() *Config {
```

Comment:
I get that calling this `Default` makes it shorter when chaining but I'm not sure I see a reason to not call this `EnabledByDefault` other than the conflicting name with the field.

If you don't want to change the field name due to the collision (which we probably shouldn't due to being exported and a breaking change) maybe `SetEnabledByDefault` which is what the function does or maybe `WithEnabledByDefault` to match the other methods on the struct?

Reply:
I don't want to use `SetEnabledByDefault` or `WithEnabledByDefault` because the function doesn't have or need args.

I switch `Default` and `EnabledByDefault` names.

Reply:
Why does it matter if it takes any arguments? `WithLoadForGoAnalysis` is named with the same pattern and takes no arguments.

Reply:
you are right, I will rename it for homogeneity.